### PR TITLE
Fix analyzer warnings in firebase_database

### PR DESCRIPTION
**Pull request by goderbauer**
See https://github.com/astashov/dartdocs.org/issues/47.

### CODE REVIEW

File: packages/firebase_database/lib/ui/firebase_list.dart [hunk 1]
```diff
@@ -47,9 +49,16 @@ class FirebaseList extends ListBase<DataSnapshot> with StreamSubscriberMixin<Eve
   /// Called when the data of the list has finished loading
   final ValueCallback onValue;
 
+  // ListBase implementation
   final List<DataSnapshot> _snapshots = <DataSnapshot>[];
   int get length => _snapshots.length;
+  set length(int value) {
+    _snapshots.length = value;
```
Comment (collinjackson) [inline packages/firebase_database/lib/ui/firebase_list.dart]: This class is intended to provide an unmodifiable view of the list. I think the solution is to extend `UnmodifiableListBase` instead, or `throw new UnsupportedError("Cannot modify an unmodifiable list");`

File: packages/firebase_database/lib/ui/firebase_list.dart [hunk 2]
```diff
@@ -47,9 +49,16 @@ class FirebaseList extends ListBase<DataSnapshot> with StreamSubscriberMixin<Eve
...
+  }
   DataSnapshot operator [](int index) => _snapshots[index];
+  void operator []=(int index, DataSnapshot value) {
+    _snapshots[index] = value;
```
Comment (collinjackson) [inline packages/firebase_database/lib/ui/firebase_list.dart]: Same here

File: packages/firebase_database/lib/ui/firebase_sorted_list.dart [hunk 3]
```diff
@@ -47,16 +49,19 @@ class FirebaseSortedList extends ListBase<DataSnapshot> with StreamSubscriberMix
...
-  final ChildMovedCallback onChildMoved;
-
   /// Called when the data of the list has finished loading
   final ValueCallback onValue;
 
   // ListBase implementation
   final List<DataSnapshot> _snapshots = <DataSnapshot>[];
   int get length => _snapshots.length;
+  set length(int value) {
+    _snapshots.length = value;
```
Comment (collinjackson) [inline packages/firebase_database/lib/ui/firebase_sorted_list.dart]: Same here.

File: packages/firebase_database/lib/ui/firebase_sorted_list.dart [hunk 4]
```diff
@@ -47,16 +49,19 @@ class FirebaseSortedList extends ListBase<DataSnapshot> with StreamSubscriberMix
...
+  }
   DataSnapshot operator [](int index) => _snapshots[index];
+  void operator []=(int index, DataSnapshot value) {
+    _snapshots[index] = value;
```
Comment (collinjackson) [inline packages/firebase_database/lib/ui/firebase_sorted_list.dart]: Same here

File: packages/firebase_database/lib/ui/firebase_list.dart (hunk 1 above)
Comment (goderbauer) [inline packages/firebase_database/lib/ui/firebase_list.dart]: I couldn't find a suitable base class. I went with the throw solution instead.

**Pull request merged**